### PR TITLE
Issue #10392: Forbid usage of deprecated API

### LINKED ISSUE #10392: Forbid usage of deprecated API Configuration.xxxAttributeXxxx
see https://github.com/checkstyle/checkstyle/issues/10164

We need to disallow usage of deprecated API

### PR DESCRIPTION
**Pull request by shashwatj07**
Resolves #10392

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1978,6 +1978,45 @@
               </execution>
             </executions>
           </plugin>
+
+          <plugin>
+            <groupId>de.thetaphi</groupId>
+            <artifactId>forbiddenapis</artifactId>
```
Comment (romani) [inline pom.xml:1984]: Can we reuse existing instance of this plugin https://github.com/checkstyle/checkstyle/blob/master/pom.xml#L1540

**Review by romani — CHANGES_REQUESTED** [review]
Items:

File: pom.xml (hunk 1 above)
Comment (shashwatj07) [reply pom.xml]: my bad, I didn't notice. fixed now.

File: forbiddenapis/signatures.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+com.puppycrawl.tools.checkstyle.DefaultConfiguration#getAttributeNames() @ Usage of deprecated API is forbidden. Please use DefaultConfiguration.getPropertyNames() instead.
```
Comment (romani) [inline forbiddenapis/signatures.txt:1]: Please move this file to "config" folder

**Review by romani — CHANGES_REQUESTED** [review]
Item before merge:

File: forbiddenapis/signatures.txt (hunk 2 above)
Comment (shashwatj07) [reply forbiddenapis/signatures.txt]: done

**Review by romani — APPROVED** [review]

**Pull request merged**